File: CodingTestPlatform/BOJ/2_Silver/5/7785/7785.py
# 회사에 있는 사람
# https://www.acmicpc.net/problem/7785

import sys
input = sys.stdin.readline

n = int(input().rstrip())
hl = set([])

for _ in range(n):
    name, state = input().split()
    if state == "enter":
        hl.add(name)
    elif state == "leave":
        hl.remove(name)

rst = list(hl)
rst.sort(reverse=True)
print('\n'.join(rst))

File: CodingTestPlatform/BOJ/2_Silver/5/7785/7785.js
/**
 * 회사에 있는 사람
 * Silver5
 * https://www.acmicpc.net/problem/7785
 * Set to Array [reference]: https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/from
 */

const [n, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split("\n");
const s = new Set([]);

input.forEach((log) => {
    const [name, state] = log.split(" ");
    if (state === 'enter') {
        s.add(name);
    } else if (state === 'leave') {
        s.delete(name);
    }
});

console.log(Array.from(s).sort().reverse().join("\n"));